lib/api/objectPutPart.js

```diff
@@ -280,14 +293,14 @@ function objectPutPart(authInfo, request, streamingV4Params, log,
                 partLocations[0].sseCipheredDataKey = cipheredDataKey;
             }
             const omVal = {
-                // Version 3 changes the format of partLocations from an object
-                // to an array
-                'md-model-version': 3,
+                // Version 4 adds number-subparts for Azure data backend
+                'md-model-version': 4,
                 partLocations,
                 'key': partKey,
                 'last-modified': new Date().toJSON(),
                 'content-md5': hexDigest,
                 'content-length': size,
+                'number-subparts': dataGetInfo.numberSubParts,
```

Comment:
@electrachong, do we need this or can we just rely on the total size?

Reply:
you can rely on total size but Arsenal mpu utils for putting subparts needs to be updated to put the correct size for the subparts (I have an in-progress branch for it)